opensrp-immunization/src/main/java/org/smartregister/immunization/domain/VaccineSchedule.java

```diff
@@ -289,6 +308,50 @@ private static List<String> getAlertsNotInNew(List<Alert> newAlerts) {
         return notInNew;
     }
 
+    public static void standardiseCalendarDate(Calendar calendarDate) {
+        calendarDate.set(Calendar.HOUR_OF_DAY, 0);
+        calendarDate.set(Calendar.MINUTE, 0);
+        calendarDate.set(Calendar.SECOND, 0);
+        calendarDate.set(Calendar.MILLISECOND, 0);
+    }
+
+    /**
+     * This method adds an offset to the provided calendar. Offsets can look like: "+5y,3m,2d" : Plus 5 years, 3 months, and
+     * 2 days "-2d" : Minus 2 days
+     * <p>
+     * Accepted time units for the offset are: d : Days m : Months y : Years
+     * <p>
+     * Accepted operators for the offset are: - : Minus + : Plus
+     *
+     * @param calendar The calendar to add the offset
+     * @param offset   The offset in the format above to add to the calendar
+     * @return
+     */
+    public static Calendar addOffsetToCalendar(Calendar calendar, String offset) {
+        if (calendar != null && offset != null) {
+            VaccinatorUtils.processConfigCalendarOffset(calendar, offset);
+        }
+
+        return calendar;
+    }
+
+    public static VaccineSchedule getVaccineSchedule(String vaccineCategory, String vaccineName) {
+        if (vaccineSchedules != null && vaccineSchedules.containsKey(vaccineCategory) && vaccineSchedules.get(vaccineCategory)
+                .containsKey(vaccineName.toUpperCase())) {
```

Comment:
There is a possibility of `vaccineName.toUpperCase()` throwing a null pointer exception